src/main/scala/com/feedzai/cosytest/DockerComposeSetup.scala

```diff
@@ -251,6 +337,43 @@ case class DockerComposeSetup(
     }
   }
 
+  /**
+    * Returns the IP of the container.
+    *
+    * @param id of the container
+    * @return an IP InetAddress option if ip was discovered. Otherwise None.
+    */
+  def getContainerIp(id: String): Option[InetAddress] = {
+    getAllNetworkNames().collectFirst {
+      case network if { getContainerIp(id, network).isDefined } => getContainerIp(id, network).get
+    }
+  }
+
+  /**
+    * Returns the IP (InetAddress) of the container.
+    *
+    * @param id of the container
+    * @param network of the container
+    * @return an IP InetAddress option if ip was discovered. Otherwise None.
+    */
+  private def getContainerIp(id: String, network: String): Option[InetAddress] = {
+    val cmd = Seq("docker", "inspect", "-f", s"""{{(index .NetworkSettings.Networks "$network").IPAddress}}""", id)
```

Comment:
I think you can just do `docker inspect <network id>`

Reply:
Yeah I didn't choose that at the time because IPv4Address is trailed by `/16` for example `"IPv4Address": "172.20.0.2/16"`.
But I guess it is easier and faster to parse it instead of fetching all network names.